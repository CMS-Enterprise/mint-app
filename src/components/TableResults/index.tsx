import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { FilterValue } from 'react-table';
import classnames from 'classnames';

type TableResultsProps = {
  className?: string;
  globalFilter: FilterValue;
  pageIndex: number;
  pageSize: number;
  filteredRowLength: number;
  rowLength: number;
};

const displayResult = (searchTerm: FilterValue) =>
  searchTerm ? (
    <span>
      <Trans i18nKey="tableAndPagination:results.searchInput" />{' '}
      <strong>&quot;{searchTerm}&quot;</strong>{' '}
    </span>
  ) : (
    ''
  );

const TableResults = ({
  className,
  globalFilter,
  pageIndex,
  pageSize,
  filteredRowLength,
  rowLength
}: TableResultsProps) => {
  const { t } = useTranslation('tableAndPagination');

  // Sets the max results to either the filtered dataset or default (depending on if filtering)
  const rows: number = globalFilter ? filteredRowLength : rowLength;

  const currentPage: number = pageIndex * pageSize + 1;

  // If data or filter results are less than 10 (page size) - then default to the number of returned rows
  const pageRange: number = rows < 10 ? rows : (pageIndex + 1) * pageSize;

  return (
    <div className={classnames(className)} data-testid="page-results">
      <span>
        {rows === 0 ? (
          <div>
            {t('tableAndPagination:results.noResults')}{' '}
            {/* Displays the search input even if there are no results */}
            {displayResult(globalFilter)}
          </div>
        ) : (
          <div>
            <Trans i18nKey="tableAndPagination:results.results">
              indexZero {{ currentPage }} indexOne {{ pageRange }} indexTwo{' '}
              {{ rows }}
            </Trans>
            {displayResult(globalFilter)}
          </div>
        )}
      </span>
    </div>
  );
};

export default TableResults;
