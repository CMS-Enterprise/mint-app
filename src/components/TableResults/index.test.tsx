import React from 'react';
import { render, screen } from '@testing-library/react';

import TableResults from './index';

describe('Table Results Componenet', () => {
  it('renders without errors', () => {
    const { getByTestId } = render(
      <TableResults
        globalFilter=""
        pageIndex={1}
        pageSize={10}
        filteredRowLength={0}
        rowLength={10}
      />
    );

    expect(getByTestId('page-results')).toBeInTheDocument();
  });

  it('display current page number of page size out of total pages', async () => {
    render(
      <TableResults
        globalFilter=""
        pageIndex={4}
        pageSize={10}
        filteredRowLength={0}
        rowLength={456}
      />
    );

    expect(
      await screen.getByText(/Showing 41-50 of 456 results/)
    ).toBeInTheDocument();
  });

  it('display filters results of page size out of total pages', async () => {
    render(
      <TableResults
        globalFilter="Happiness"
        pageIndex={0}
        pageSize={10}
        filteredRowLength={42}
        rowLength={456}
      />
    );

    expect(
      await screen.getByText(/Showing 1-10 of 42 results/)
    ).toBeInTheDocument();
    expect(await screen.findByText('"Happiness"')).toBeInTheDocument();
  });

  it('matches snapshot', async () => {
    const { asFragment } = render(
      <TableResults
        globalFilter=""
        pageIndex={0}
        pageSize={10}
        filteredRowLength={0}
        rowLength={100}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
