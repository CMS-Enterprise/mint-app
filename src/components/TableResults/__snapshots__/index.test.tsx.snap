// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Results Componenet matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="page-results"
  >
    <span>
      <div>
        Showing 1-10 of 100 results 
      </div>
    </span>
  </div>
</DocumentFragment>
`;
